src/main.rs

```diff
@@ -49,4 +81,67 @@ fn main() {
         gelf_reader,
     ));
     system.run().unwrap();
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+
+    use std::env;
+    use std::io::Write;
+    use tempfile::NamedTempFile;
+
+    // make sure to run tests in one thread only, otherwise it is possible we get race conditions for
+    // environment variables set/unset and assigning both file tests the same filename
+    use serial_test::serial;
```

Comment:
Might also be possible to do this with, `cargo test -j 1` but since it's a dev dependency I think it's not super critical.

Reply:
I've used the option with only one thread before, but you always have the problem how to document you need this. And in case you have more tests you can run only selected tests serial instead of all.